src/foam/net/node/RestDAOHandler.js

```diff
@@ -0,0 +1,221 @@
+        }).then(function() {
+          self.sendJSON(res, 200, payload);
+          self.info('200 OK: remove() ' + id);
+        }).catch(send500);
+      } else if ( req.method === 'GET' ) {
+        if ( sep === '/' && target !== '' ) {
+          // Extra fragment: "/<id>" => find().
+
+          //
+          // find()
+          //
+
+          try {
+            id = JSON.parse(decodeURIComponent(target));
+          } catch (err) {
+            send500(err);
+            return true;
+          }
+
+          self.dao.find(id).then(function(o) {
+            self.sendJSON(res, 200, self.fo2o_(o));
+            self.info('200 OK: find() ' + id);
+          }).catch(send500);
+        } else if ( sep === ':' && target === 'select' ) {
+          // Extra fragment: ":select" => select().
+
+          //
+          // select()
+          //
+
+          // Decode query. E.g., skip=<skip>&predicate=<predicate>
+          // Values in key=value are wrapped in
+          // uri-encode(json-stringify(foam-jsonify(value))).
+          data = url.query;
+          try {
+            for ( var key in data ) {
+              // Note: This would use data.hasOwnProperty(key), except that
+              // Node JS ParsedQueryString objects do not descend from
+              // Object.prototype.
+              data[key] = self.jsonStr2fo_(decodeURIComponent(data[key]));
+            }
+          } catch (error) {
+            send500(error);
+            self.error('Failed to decode query from URL fragment: ' + target);
+            return true;
+          }
+
+          var skip = data.skip;
+          var limit = data.limit;
+          var order = data.order;
+          var predicate = data.predicate;
+          self.dao.select(undefined, skip, limit, order, predicate)
+            .then(function(sink) {
+              self.sendJSON(res, 200, self.fo2o_(sink.a));
+              self.info('200 OK: select() ' + sink.a.length);
+            }).catch(send500);
+        } else {
+          self.send404(req, res);
+          self.warn('Unrecognized DAO GET URL fragment: '  + sep + target);
+        }
+      } else if ( req.method === 'POST' ) {
+        if ( sep !== ':' || target !== 'removeAll' ) {
+          self.send404(req, res);
+          self.warn('Unknown POST request: ' + target);
+          return true;
+        }
+
+
+        //
+        // removeAll()
+        //
+
+        self.getPayload_(req).then(function(data) {
+          var skip = data.skip;
+          var limit = data.limit;
+          var order = data.order;
+          var predicate = data.predicate;
+          return self.dao.removeAll(skip, limit, order, predicate);
+        }).then(function() {
+          self.sendJSON(res, 200, '');
+          self.info('200 OK: removeAll()');
+        }).catch(send500);
+      } else {
+        self.send404(req, res);
+        self.warn('Method not supported: '  + req.method);
+      }
+
+      return true;
+    },
+    {
+      name: 'fo2o_',
```

Comment:
It does make sense, just pointing out that it's inconsistent. I'm fine either way.